doc/dev/python_scheduling/autotune_inner_reduction.py

```diff
@@ -0,0 +1,395 @@
+
+    def __repr__(self):
+        return f"inner_reduction_{self.selected_fusion.name}"
+
+    def convert_to_inner_reduction_params(self, scheduler_config, reduction_params):
+        warp_size = 32
+        max_number_of_threads_cta = 1024
+        grid_x_limit = 2147483647
+        grid_y_limit = 65535
+
+        reduction_params.schedule_3D = False
+        reduction_params.fastest_dim = True
+        reduction_params.cross_block_inner_reduction = True
+        reduction_params.block_dim_inner_reduction = ParallelType.block_x
+        reduction_params.cross_grid_inner_reduction = scheduler_config.grdim > 1
+        reduction_params.multiple_reds_per_blk = scheduler_config.bdimy > 1
+        reduction_params.pad_inner_reduction_to_warp = (
+            scheduler_config.bdimx > warp_size
+        ) and (
+            (scheduler_config.bdimx * scheduler_config.bdimy)
+            < max_number_of_threads_cta
+        )
+        reduction_params.unroll_factor_inner_reduction = (
+            scheduler_config.vectorize_factor
+        )
+        reduction_params.vectorize_inner_reduction = (
+            scheduler_config.vectorize_factor > 1
+        )
+
+        if scheduler_config.bdimy > 1:
+            reduction_params.block_dim_iter_dom = ParallelType.block_y
+
+        reduction_params.unroll_factor_iter_dom = scheduler_config.unroll_factor
+
+        gdimx = -1
+        gdimy = -1
+
+        if scheduler_config.grdim > 1:
+            reduction_params.grid_dim_inner_reduction = ParallelType.grid_x
+            reduction_params.grid_dim_iter_dom = ParallelType.grid_y
+
+            reduction_params.split_grid_dim_iter_dom_inner = True
+            gdimx = min(scheduler_config.grdim, grid_x_limit)
+            gdimy = min(scheduler_config.godim, grid_y_limit)
+            if scheduler_config.godim > grid_y_limit:
+                reduction_params.split_grid_dim_iter_dom_outer = True
+        else:
+            reduction_params.grid_dim_iter_dom = ParallelType.grid_x
+            gdimx = min(scheduler_config.godim, grid_x_limit)
+            if scheduler_config.godim > grid_x_limit:
+                reduction_params.split_grid_dim_inner_reduction = True
+
+        reduction_params.lparams.gdimx = gdimx
+        reduction_params.lparams.gdimy = gdimy
+
+        # Reset CTA dimensions to avoid failing LaunchParams::assertValid
+        reduction_params.lparams.bdimx = -1
+        reduction_params.lparams.bdimy = -1
+        reduction_params.lparams.bdimz = -1
+
+        reduction_params.lparams.bdimx = scheduler_config.bdimx
+        reduction_params.lparams.bdimy = scheduler_config.bdimy
+
+    # For reduction scheduler, we test the cartesian product of vectorization and
+    # unroll factors.
+    def generate_scheduler_configurations(self, input_shape):
+        threads_per_cta_options = [128, 256, 512, 1024]
+        vectorization_factor_options = [1, 2, 4, 8]
+        unroll_factor_options = list(range(1, 11))
```

Comment:
we have `iteration unroll` and `reduction unroll on top of vectorization`

Reply:
I'll add it in a separate PR.

Reply:
https://github.com/NVIDIA/Fuser/pull/3487